### PR TITLE
Adds Keycloak OAuth provider

### PR DESCRIPTION
**Pull request by infomiho**
- Adds the Keycloak OAuth provider using Arctic. 
- It extends all of the place where we had Google and Github to include Keycloak (icons, buttons, Wasp Studio etc.)

### Left to do
- [x] Update the docs to include Keycloak info
  ```ts
    [ Server ] {
    [ Server ]   profile: {
    [ Server ]     sub: '5adba8fc-3ea6-445a-a379-13f0bb0b6969',
    [ Server ]     email_verified: true,
    [ Server ]     name: 'Test User',
    [ Server ]     preferred_username: 'test',
    [ Server ]     given_name: 'Test',
    [ Server ]     family_name: 'User',
    [ Server ]     email: 'test@example.com'
    [ Server ]   }
    [ Server ] }
  ```
- [x] Update Keycloak impl to match the latest impl of other providers

### CODE REVIEW

File: waspc/data/Generator/templates/server/src/auth/providers/config/keycloak.ts
Comment (sodic) [inline waspc/data/Generator/templates/server/src/auth/providers/config/keycloak.ts:1]: I don't have any specific comments (besides the discussion we're having here: https://github.com/wasp-lang/wasp/pull/1851#discussion_r1516682178

File: web/docs/auth/social-auth/keycloak.md
Comment (sodic) [inline web/docs/auth/social-auth/keycloak.md:86]:
```suggestion
The `userEntity` is explained in [the social auth overview](../../auth/social-auth/overview#social-login-entity).
```

File: web/docs/auth/social-auth/keycloak.md
Comment (sodic) [inline web/docs/auth/social-auth/keycloak.md:123]:
```suggestion
1. Log into your Keycloak admin console.
```

File: web/docs/auth/social-auth/keycloak.md
Comment (sodic) [inline web/docs/auth/social-auth/keycloak.md:128]:
```suggestion
1. Fill in the "Client ID" and choose a name for the client.
```

File: web/docs/auth/social-auth/keycloak.md
Comment (sodic) [inline web/docs/auth/social-auth/keycloak.md:249]: I think you should import `React`.

File: web/docs/auth/social-auth/keycloak.md
Comment (sodic) [inline web/docs/auth/social-auth/keycloak.md:331]:
```suggestion
We are using Keycloak's API and its `/userinfo` endpoint to fetch the user's data.
```

File: web/docs/auth/social-auth/keycloak.md
Comment (sodic) [inline web/docs/auth/social-auth/keycloak.md:124]: In English, the quotation marks go inside the quotation marks:

```suggestion
1. Under "Clients," click on "Create Client."
```

It doesn't make any sense, I know, but that's the rule.

File: web/docs/auth/social-auth/keycloak.md
Comment (sodic) [inline web/docs/auth/social-auth/keycloak.md:203]:
```suggestion
Let's now create an `auth.{jsx,tsx}` file in the `src/pages`.
```

File: web/docs/auth/social-auth/keycloak.md
Comment (sodic) [inline web/docs/auth/social-auth/keycloak.md:512]:
```suggestion
The `keycloak` dictionary has the following properties:
```

**Review by sodic — COMMENTED** [review]
All good, found only a couple of small docs errors to correct (and some suggestions you can consider).

Other than that, LGTM.

File: web/docs/auth/social-auth/keycloak.md
Comment (infomiho) [reply web/docs/auth/social-auth/keycloak.md:512]: We call the type in Wasp language `dict` https://wasp-lang.dev/docs/general/language#complete-list-of-wasp-types 🤷

File: web/docs/auth/social-auth/keycloak.md
Comment (infomiho) [reply web/docs/auth/social-auth/keycloak.md:249]: Types work just fine even without importing it, I've just tested it

File: web/docs/auth/social-auth/keycloak.md
Comment (infomiho) [reply web/docs/auth/social-auth/keycloak.md]: I've decided to bold it instead of putting it in quotes. It avoids this situation and that's what we did in the other guides.

**Comment by infomiho** [comment]
I'll merge this after we merge the Google and Github changes and docs.

File: web/docs/auth/social-auth/keycloak.md
Comment (sodic) [reply web/docs/auth/social-auth/keycloak.md:249]: Hm, strange. But if everything works (hope you checked build), then no problem.

I'd still probably do:
```typescript
import { ReactNode } from 'react';
```
But it's just a matter of personal style.

File: web/docs/auth/social-auth/keycloak.md
Comment (sodic) [reply web/docs/auth/social-auth/keycloak.md:512]: Ha, didn't know that!

I always thought the author was lazy whenever I read `dict`.

**Review by sodic — APPROVED** [review]
LGTM

**Pull request merged**
